mrblib/compar.rb

```diff
@@ -48,7 +48,11 @@ def <= other
   #
   # ISO 15.3.3.2.3
   def == other
+    @@eq_call_stack ||= []
+    return false if @@eq_call_stack.include?([self.object_id, other.object_id])
```

Comment:
I am afraid that this recursive comparison check may consume too much time and memory.
Do we really need this?

Reply:
Hmm, let me think about the issue. The cost is not preferable.